### PR TITLE
Rimworld open world

### PR DESCRIPTION
**Pull request by Vaultpls**
Added Rimworld Open World Server mod that uses Debian 11 with usage of DotNET3.1 Core

# Description

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [x] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [x] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [x] Have you added a unique README.md for the egg you are adding according to the alphabetical order?

### CODE REVIEW

**Comment by QuintenQVD0** [comment]
Wy do you even install dotnet in the install container? and you use a debian image and export dotnet root? wy not just use the dotnet image?

**Comment by Vaultpls** [comment]
> Wy do you even install dotnet in the install container? and you use a debian image and export dotnet root? wy not just use the dotnet image?

Good question.  I didn't actively search for the solution to my problem so I manually did it 😅

**Comment by QuintenQVD0** [comment]
Have a look here: https://github.com/QuintenQVD0/Q_eggs/blob/main/egg-open-world.json I already have a sort of working one but with arm64 support. Feel free to base you on this.

File: game_eggs/rimworld/open_world/egg-rimworld-open-world.json [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    "exported_at": "2022-12-01T15:59:26+00:00",
+    "name": "Open World",
+    "author": "kyle@williquette.us",
+    "description": "The Open World mod allows you to create and join servers to play with people all around the world, it introduces plenty of new features including trading, gifting, bartering, PvP and so much more!\r\n\r\nhttps:\/\/github.com\/TastyLollipop\/OpenWorld",
+    "features": null,
+    "docker_images": {
+        "ghcr.io\/parkervcp\/yolks:debian": "ghcr.io\/parkervcp\/yolks:debian"
+    },
+    "file_denylist": [],
+    "startup": "DOTNET_ROOT=\/home\/container;\/home\/container\/\"Open World Server\"",
```
Comment (parkervcp) [inline game_eggs/rimworld/open_world/egg-rimworld-open-world.json:16]: This could be an egg ENV instead `DOTNET_ROOT=\/home\/container`

**Review by parkervcp — CHANGES_REQUESTED** [review]

File: game_eggs/rimworld/open_world/egg-rimworld-open-world.json (hunk 1 above)
Comment (Vaultpls) [reply game_eggs/rimworld/open_world/egg-rimworld-open-world.json]: Just made the change and did a fix to the max_player from being limited to 3 max to now allow unlimited players(oops)

**Comment by Vaultpls** [comment]
Also if anyone wanted to try out the test/production server regarding this, I am hosting one at game.theboys.gg @ port 25555.

Server requirements is very light(less than 500mb RAM for 5 players online) and disk was less than 200MB

Feel free to DM/remove this isn't allowed.

**Comment by QuintenQVD0** [comment]
You are setting dotnet root on a server that the docker image does not even have dotnet?
and you are installing dotnet in the install container? just use a prebuild one

File: game_eggs/rimworld/open_world/egg-rimworld-open-world.json [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+            "container": "ghcr.io\/parkervcp\/yolks:debian",
```
Comment (parkervcp) [inline game_eggs/rimworld/open_world/egg-rimworld-open-world.json:26]: You should only need `ghcr.io/parkervcp/installers:debian` to run the install.

You can also skip adding dotnet and other things during the install.

**Review by parkervcp — CHANGES_REQUESTED** [review]

**Comment by QuintenQVD0** [comment]
I rewrote this egg as they rewrote the aplication

**Review by QuintenQVD0 — APPROVED** [review]

**Comment by saffue** [comment]
Hey! thanks for this egg, much better than the super basic one i made lol. we updated less than a week ago and changed the config file to include a few more lines, specifically these, is it possible that they get added at some point, thanks<3
```
  "BackUpIntervalInHours": 24,
  "DevConsole": true,
  "ServerPassword": "",
```

**Comment by QuintenQVD0** [comment]
> Hey! thanks for this egg, much better than the super basic one i made lol. we updated less than a week ago and changed the config file to include a few more lines, specifically these, is it possible that they get added at some point, thanks<3
> 
> ```
>   "BackUpIntervalInHours": 24,
>   "DevConsole": true,
>   "ServerPassword": "",
> ```

does the offical config now come with those lines?

**Comment by saffue** [comment]
i will have a look into it, tho never made a config for one

**Pull request merged**